simple/utils/spectra_convert.py

```diff
@@ -56,81 +59,59 @@ def compile_header(wavelength_data, **original_header_dict):
     header = fits.Header()
     header.set("EXTNAME", "PRIMARY", "name of this extension")
 
-    try:
-        header.set("VOCLASS", original_header_dict["voclass"], "VO Data Model")
-    except KeyError:
-        logging.warning("No VO Data Model")
+    # try:
+    #    header.set("VOCLASS", original_header_dict["voclass"], "VO Data Model")
+    # except KeyError:
+    #    logging.warning("No VO Data Model")
 
-    # Target Data
+    # REQUIRED Target Data
     try:
         header.set(
             "OBJECT", original_header_dict["object_name"], "Name of observed object"
         )
     except KeyError:
-        logging.warning("No object name")
-
-    try:
-        ra = original_header_dict["RA"]
-        header.set("RA", ra, "[deg] Pointing position")
-    except KeyError:
-        logging.warning("No RA")
-        ra = None
-    try:
-        dec = original_header_dict["dec"]
-        header.set("DEC", dec, "[deg] Pointing position")
-    except KeyError:
-        logging.warning("No DEC")
-        dec = None
-
-    try:
-        time = (
-            Time(to_datetime(original_header_dict["start_time"])).jd
-            + Time(to_datetime(original_header_dict["stop_time"])).jd
-        ) / 2
-        header.set("TMID", time, "[d] MJD mid expsoure")
-    except KeyError:
-        time = None
-        logging.warning("No time")
-
-    try:
-        exposure_time = original_header_dict["exposure_time"]
-        header.set("TELAPSE", exposure_time, "[s] Total elapsed time")
-    except KeyError:
-        exposure_time = None
-        logging.warning("No exposure time")
+        object_name = input("REQUIRED: Please input a name for the object: ")
```

Comment:
I'm not a fan of the input function for several reasons, but here are two main points:

1. Testing this function is now more complex- you have to simulate a variable number of user inputs and while there are surely methods in pytest to do so, it'll require more setup.
2. The flow of this function is not at all consistent. If values exist, no inputs are needed, sometimes some inputs are needed but not others. Automating this, if you want to loop through a list, will become nearly impossible.

The original implementation, which I see you commented out around line 50 is better (though can be tweaked)- identify all missing keywords from your input dictionary and fail if any of them are missing.

Reply:
The target user of this function (at the moment) is a high school student or undergrad who is using the spectrum for a different purpose. I want a function which will hold their hand through putting together our highest priority header keywords. 
- I have used the `input()`  function for what I consider the highest priority keywords.
- For keywords which they can likely figure out, e.g., Instrument , I've also used the `input()` function but labeled them as OPTIONAL.
- for keywords which may have existed in the original header, I've just pulled them from the original header with no user interaction. 

If you have another suggestion for how to do the hand holding, I'm open!

I also wasn't planning on doing too much testing in our CI...I was gonna let the students do the testing. As they find things which don't work, we can modify the function and add tests? 

I also recognize that this function is not mission critical to SIMPLE -- the database will still work with heterogenous spectra files -- so I'm ok if this function is not to the same standard as our ingest code.

Reply:
all that being said, this could be a useful script in `astrodb_utils` at some point, especially since it's striving to be consistent with SpectrumDM 1.2.  We could rename it `compile_ivoa_spectrum_dm_header`.  And I'm working on doing the same thing for light curves at the minute.  So maybe making it high quality code is a good idea....

Reply:
One thing that occurs to me is that we can split this into two (or more) functions: 
1. one function that takes a dictionary and compiles the header and 
2. one that validates and creates the dictionary

That later function can be one using the `input()` calls (and we can add helper functions if needed), but the compile header function is more streamlined and can be tested or automated more readily.

Reply:
I think two functions is a very good idea. I think I may also go ahead and develop this in `astrodb_utils` instead of here in SIMPLE.